src/components/DetailsTable.js

```diff
@@ -28,8 +28,14 @@ const DetailsTable = (props) => {
     const header = props.tableHeader
 
     const rows = props.data.map((data)=> _.pick(data, props.allowedKeys));
+
+    const renderColumns = row => {
+        const columnAccessors = Object.keys(row);
+        // This uses the object keys to iterate through the columns per row.
+        return columnAccessors.map(column => <TableCell align="center" className={classes.cellRow}>{row[column]}</TableCell>)
+    };
```

Comment:
There's a lot to unpack here, but we're dynamically rendering table rows and columns without being specific or care about the object that's being passed by double iterations (.map() in this case).

A table can be represented in JSON by an array of objects with standard keys. Each object in the array is a row, and each property on the object is a column.

Example:
```js
[
  { columnName: 'Cell Value', secondColumnName: 'cell Value' },
  { columnName: 'Cell Value', secondColumnName: 'cell Value' },
]
```

Then, you essentially do a loop within a loop. :)  

**First loop:** iterating over the Table Rows. Easy enough.
**Second Loop:** As we iterate over each row, we ALSO iterate over its columns (the keys in a row object). This builds out the columns for that given row.